source/extensions/clusters/redis/redis_cluster.h

```diff
@@ -214,10 +214,10 @@ class RedisCluster : public Upstream::BaseDynamicClusterImpl {
     uint32_t maxBufferSizeBeforeFlush() const override { return 0; }
     std::chrono::milliseconds bufferFlushTimeoutInMs() const override { return buffer_timeout_; }
     uint32_t maxUpstreamUnknownConnections() const override { return 0; }
-    // This is effectively not in used for making the "Cluster Slots" calls.
-    // since we call cluster slots on both the master and slaves, ANY is more appropriate here.
+    // This is effectively not in used for making the "Cluster Slots" calls, using Master to avoid
+    // the READONLY command when establishing connection.
```

Comment:
Could you flesh this comment out more? I don't understand from the comment why this was changed from any to master and what this is actually used for?

Reply:
Before the change, the RedisDiscoverySession don't send "READONLY" and this readPolicy was not used at all in this context.  After the change, the code to send "READONLY" command when a connection is created is move to RedisClientImpl, and it's gated on the readPolicy, so we need to set it to Master here to have the same behavior as before.